Adds free form tags support. Remove this once this is added in upstream SDK

diff --git a/github.com/oracle/oci-go-sdk/cmd/genver/main.go b/github.com/oracle/oci-go-sdk/cmd/genver/main.go
index d9ba965..8ecf415 100644
--- a/github.com/oracle/oci-go-sdk/cmd/genver/main.go
+++ b/github.com/oracle/oci-go-sdk/cmd/genver/main.go
@@ -42,7 +42,7 @@ func main() {
 		Tag                 string
 	}{
 		getEnvOrDefault(majorVer, "0"),
-		getEnvOrDefault(minorVer, "0" ),
+		getEnvOrDefault(minorVer, "0"),
 		getEnvOrDefault(patchVer, "0"),
 		getEnvOrDefault(tag, ""),
 	}
diff --git a/github.com/oracle/oci-go-sdk/cmd/genver/version_template.go b/github.com/oracle/oci-go-sdk/cmd/genver/version_template.go
index 5c684fd..f9fc03c 100644
--- a/github.com/oracle/oci-go-sdk/cmd/genver/version_template.go
+++ b/github.com/oracle/oci-go-sdk/cmd/genver/version_template.go
@@ -1,6 +1,5 @@
 package main
 
-
 const versionTemplate = `
 // Copyright (c) 2016, 2018, Oracle and/or its affiliates. All rights reserved.
 // Code generated by go generate; DO NOT EDIT
@@ -39,4 +38,3 @@ func Version() string {
 	return version
 }
 `
-
diff --git a/github.com/oracle/oci-go-sdk/core/create_internet_gateway_details.go b/github.com/oracle/oci-go-sdk/core/create_internet_gateway_details.go
index df49670..910f36d 100644
--- a/github.com/oracle/oci-go-sdk/core/create_internet_gateway_details.go
+++ b/github.com/oracle/oci-go-sdk/core/create_internet_gateway_details.go
@@ -26,6 +26,9 @@ type CreateInternetGatewayDetails struct {
 
 	// A user-friendly name. Does not have to be unique, and it's changeable. Avoid entering confidential information.
 	DisplayName *string `mandatory:"false" json:"displayName"`
+
+	// Free form tags applied to this resource
+	FreeFormTags map[string]string `mandatory:"false" json:"freeformTags,omitempty"`
 }
 
 func (m CreateInternetGatewayDetails) String() string {
diff --git a/github.com/oracle/oci-go-sdk/core/create_route_table_details.go b/github.com/oracle/oci-go-sdk/core/create_route_table_details.go
index 8f3005a..9fb3bfc 100644
--- a/github.com/oracle/oci-go-sdk/core/create_route_table_details.go
+++ b/github.com/oracle/oci-go-sdk/core/create_route_table_details.go
@@ -26,6 +26,9 @@ type CreateRouteTableDetails struct {
 
 	// A user-friendly name. Does not have to be unique, and it's changeable. Avoid entering confidential information.
 	DisplayName *string `mandatory:"false" json:"displayName"`
+
+	// Free form tags applied to this resource
+	FreeFormTags map[string]string `mandatory:"false" json:"freeformTags,omitempty"`
 }
 
 func (m CreateRouteTableDetails) String() string {
diff --git a/github.com/oracle/oci-go-sdk/core/create_security_list_details.go b/github.com/oracle/oci-go-sdk/core/create_security_list_details.go
index 06d4c4e..1ddcfcc 100644
--- a/github.com/oracle/oci-go-sdk/core/create_security_list_details.go
+++ b/github.com/oracle/oci-go-sdk/core/create_security_list_details.go
@@ -29,6 +29,9 @@ type CreateSecurityListDetails struct {
 
 	// A user-friendly name. Does not have to be unique, and it's changeable. Avoid entering confidential information.
 	DisplayName *string `mandatory:"false" json:"displayName"`
+
+	// Free form tags applied to this resource
+	FreeFormTags map[string]string `mandatory:"false" json:"freeformTags,omitempty"`
 }
 
 func (m CreateSecurityListDetails) String() string {
diff --git a/github.com/oracle/oci-go-sdk/core/create_subnet_details.go b/github.com/oracle/oci-go-sdk/core/create_subnet_details.go
index bc50b2a..b97d566 100644
--- a/github.com/oracle/oci-go-sdk/core/create_subnet_details.go
+++ b/github.com/oracle/oci-go-sdk/core/create_subnet_details.go
@@ -69,6 +69,9 @@ type CreateSubnetDetails struct {
 	// the subnet. Remember that security lists are associated at the subnet
 	// level, but the rules are applied to the individual VNICs in the subnet.
 	SecurityListIds []string `mandatory:"false" json:"securityListIds"`
+
+	// Free form tags applied to this resource
+	FreeFormTags map[string]string `mandatory:"false" json:"freeformTags,omitempty"`
 }
 
 func (m CreateSubnetDetails) String() string {
diff --git a/github.com/oracle/oci-go-sdk/core/create_vcn_details.go b/github.com/oracle/oci-go-sdk/core/create_vcn_details.go
index 70eb2c2..ad9bfca 100644
--- a/github.com/oracle/oci-go-sdk/core/create_vcn_details.go
+++ b/github.com/oracle/oci-go-sdk/core/create_vcn_details.go
@@ -38,6 +38,9 @@ type CreateVcnDetails struct {
 	// DNS in Your Virtual Cloud Network (https://docs.us-phoenix-1.oraclecloud.com/Content/Network/Concepts/dns.htm).
 	// Example: `vcn1`
 	DnsLabel *string `mandatory:"false" json:"dnsLabel"`
+
+	// Free form tags applied to this resource
+	FreeFormTags map[string]string `mandatory:"false" json:"freeformTags,omitempty"`
 }
 
 func (m CreateVcnDetails) String() string {
diff --git a/github.com/oracle/oci-go-sdk/core/create_volume_details.go b/github.com/oracle/oci-go-sdk/core/create_volume_details.go
index 492c815..9b2f16d 100644
--- a/github.com/oracle/oci-go-sdk/core/create_volume_details.go
+++ b/github.com/oracle/oci-go-sdk/core/create_volume_details.go
@@ -43,6 +43,9 @@ type CreateVolumeDetails struct {
 	// This field is deprecated. Use the sourceDetails field instead to specify the
 	// backup for the volume.
 	VolumeBackupId *string `mandatory:"false" json:"volumeBackupId"`
+
+	// Free form tags applied to this resource
+	FreeFormTags map[string]string `mandatory:"false" json:"freeformTags,omitempty"`
 }
 
 func (m CreateVolumeDetails) String() string {
@@ -59,6 +62,7 @@ func (m *CreateVolumeDetails) UnmarshalJSON(data []byte) (e error) {
 		VolumeBackupId     *string             `json:"volumeBackupId"`
 		AvailabilityDomain *string             `json:"availabilityDomain"`
 		CompartmentId      *string             `json:"compartmentId"`
+		FreeFormTags       map[string]string   `json:"freeformTags"`
 	}{}
 
 	e = json.Unmarshal(data, &model)
@@ -76,5 +80,6 @@ func (m *CreateVolumeDetails) UnmarshalJSON(data []byte) (e error) {
 	m.VolumeBackupId = model.VolumeBackupId
 	m.AvailabilityDomain = model.AvailabilityDomain
 	m.CompartmentId = model.CompartmentId
+	m.FreeFormTags = model.FreeFormTags
 	return
 }
diff --git a/github.com/oracle/oci-go-sdk/core/instance.go b/github.com/oracle/oci-go-sdk/core/instance.go
index afad798..bd83a9b 100644
--- a/github.com/oracle/oci-go-sdk/core/instance.go
+++ b/github.com/oracle/oci-go-sdk/core/instance.go
@@ -86,6 +86,9 @@ type Instance struct {
 
 	// Details for creating an instance
 	SourceDetails InstanceSourceDetails `mandatory:"false" json:"sourceDetails"`
+
+	// Free form tags applied to this resource
+	FreeFormTags map[string]string `mandatory:"false" json:"freeformTags,omitempty"`
 }
 
 func (m Instance) String() string {
@@ -108,6 +111,7 @@ func (m *Instance) UnmarshalJSON(data []byte) (e error) {
 		Region             *string                    `json:"region"`
 		Shape              *string                    `json:"shape"`
 		TimeCreated        *common.SDKTime            `json:"timeCreated"`
+		FreeFormTags       map[string]string          `json:"freeformTags"`
 	}{}
 
 	e = json.Unmarshal(data, &model)
@@ -131,6 +135,7 @@ func (m *Instance) UnmarshalJSON(data []byte) (e error) {
 	m.Region = model.Region
 	m.Shape = model.Shape
 	m.TimeCreated = model.TimeCreated
+	m.FreeFormTags = model.FreeFormTags
 	return
 }
 
diff --git a/github.com/oracle/oci-go-sdk/core/internet_gateway.go b/github.com/oracle/oci-go-sdk/core/internet_gateway.go
index a9b8f1c..b602800 100644
--- a/github.com/oracle/oci-go-sdk/core/internet_gateway.go
+++ b/github.com/oracle/oci-go-sdk/core/internet_gateway.go
@@ -43,6 +43,9 @@ type InternetGateway struct {
 	// The date and time the Internet Gateway was created, in the format defined by RFC3339.
 	// Example: `2016-08-25T21:10:29.600Z`
 	TimeCreated *common.SDKTime `mandatory:"false" json:"timeCreated"`
+
+	// Free form tags applied to this resource
+	FreeFormTags map[string]string `mandatory:"false" json:"freeformTags,omitempty"`
 }
 
 func (m InternetGateway) String() string {
diff --git a/github.com/oracle/oci-go-sdk/core/launch_instance_details.go b/github.com/oracle/oci-go-sdk/core/launch_instance_details.go
index c40c778..956d77f 100644
--- a/github.com/oracle/oci-go-sdk/core/launch_instance_details.go
+++ b/github.com/oracle/oci-go-sdk/core/launch_instance_details.go
@@ -125,6 +125,9 @@ type LaunchInstanceDetails struct {
 	// CreateVnicDetails.
 	// At least one of them is required; if you provide both, the values must match.
 	SubnetId *string `mandatory:"false" json:"subnetId"`
+
+	// Free form tags applied to this resource
+	FreeFormTags map[string]string `mandatory:"false" json:"freeformTags,omitempty"`
 }
 
 func (m LaunchInstanceDetails) String() string {
@@ -146,6 +149,7 @@ func (m *LaunchInstanceDetails) UnmarshalJSON(data []byte) (e error) {
 		AvailabilityDomain *string                `json:"availabilityDomain"`
 		CompartmentId      *string                `json:"compartmentId"`
 		Shape              *string                `json:"shape"`
+		FreeFormTags       map[string]string      `json:"freeformTags"`
 	}{}
 
 	e = json.Unmarshal(data, &model)
@@ -168,5 +172,6 @@ func (m *LaunchInstanceDetails) UnmarshalJSON(data []byte) (e error) {
 	m.AvailabilityDomain = model.AvailabilityDomain
 	m.CompartmentId = model.CompartmentId
 	m.Shape = model.Shape
+	m.FreeFormTags = model.FreeFormTags
 	return
 }
diff --git a/github.com/oracle/oci-go-sdk/core/route_table.go b/github.com/oracle/oci-go-sdk/core/route_table.go
index f255d13..d4b1cfc 100644
--- a/github.com/oracle/oci-go-sdk/core/route_table.go
+++ b/github.com/oracle/oci-go-sdk/core/route_table.go
@@ -42,6 +42,9 @@ type RouteTable struct {
 	// The date and time the route table was created, in the format defined by RFC3339.
 	// Example: `2016-08-25T21:10:29.600Z`
 	TimeCreated *common.SDKTime `mandatory:"false" json:"timeCreated"`
+
+	// Free form tags applied to this resource
+	FreeFormTags map[string]string `mandatory:"false" json:"freeformTags,omitempty"`
 }
 
 func (m RouteTable) String() string {
diff --git a/github.com/oracle/oci-go-sdk/core/security_list.go b/github.com/oracle/oci-go-sdk/core/security_list.go
index d55bcd2..0afbdf8 100644
--- a/github.com/oracle/oci-go-sdk/core/security_list.go
+++ b/github.com/oracle/oci-go-sdk/core/security_list.go
@@ -50,6 +50,9 @@ type SecurityList struct {
 
 	// The OCID of the VCN the security list belongs to.
 	VcnId *string `mandatory:"true" json:"vcnId"`
+
+	// Free form tags applied to this resource
+	FreeFormTags map[string]string `mandatory:"false" json:"freeformTags,omitempty"`
 }
 
 func (m SecurityList) String() string {
diff --git a/github.com/oracle/oci-go-sdk/core/subnet.go b/github.com/oracle/oci-go-sdk/core/subnet.go
index d765cef..ebccf03 100644
--- a/github.com/oracle/oci-go-sdk/core/subnet.go
+++ b/github.com/oracle/oci-go-sdk/core/subnet.go
@@ -97,6 +97,9 @@ type Subnet struct {
 	// The date and time the subnet was created, in the format defined by RFC3339.
 	// Example: `2016-08-25T21:10:29.600Z`
 	TimeCreated *common.SDKTime `mandatory:"false" json:"timeCreated"`
+
+	// Free form tags applied to this resource
+	FreeFormTags map[string]string `mandatory:"false" json:"freeformTags,omitempty"`
 }
 
 func (m Subnet) String() string {
diff --git a/github.com/oracle/oci-go-sdk/core/update_volume_details.go b/github.com/oracle/oci-go-sdk/core/update_volume_details.go
index e0b642b..87114e7 100644
--- a/github.com/oracle/oci-go-sdk/core/update_volume_details.go
+++ b/github.com/oracle/oci-go-sdk/core/update_volume_details.go
@@ -18,6 +18,9 @@ type UpdateVolumeDetails struct {
 	// A user-friendly name. Does not have to be unique, and it's changeable.
 	// Avoid entering confidential information.
 	DisplayName *string `mandatory:"false" json:"displayName"`
+
+	// Free form tags applied to this resource
+	FreeFormTags map[string]string `mandatory:"false" json:"freeformTags,omitempty"`
 }
 
 func (m UpdateVolumeDetails) String() string {
diff --git a/github.com/oracle/oci-go-sdk/core/vcn.go b/github.com/oracle/oci-go-sdk/core/vcn.go
index 8077a53..173adcc 100644
--- a/github.com/oracle/oci-go-sdk/core/vcn.go
+++ b/github.com/oracle/oci-go-sdk/core/vcn.go
@@ -67,6 +67,9 @@ type Vcn struct {
 	// DNS in Your Virtual Cloud Network (https://docs.us-phoenix-1.oraclecloud.com/Content/Network/Concepts/dns.htm).
 	// Example: `vcn1.oraclevcn.com`
 	VcnDomainName *string `mandatory:"false" json:"vcnDomainName"`
+
+	// Free form tags applied to this resource
+	FreeFormTags map[string]string `mandatory:"false" json:"freeformTags,omitempty"`
 }
 
 func (m Vcn) String() string {
diff --git a/github.com/oracle/oci-go-sdk/core/volume.go b/github.com/oracle/oci-go-sdk/core/volume.go
index 85c7394..2d1882a 100644
--- a/github.com/oracle/oci-go-sdk/core/volume.go
+++ b/github.com/oracle/oci-go-sdk/core/volume.go
@@ -53,6 +53,9 @@ type Volume struct {
 	// The volume source, either an existing volume in the same Availability Domain or a volume backup.
 	// If null, an empty volume is created.
 	SourceDetails VolumeSourceDetails `mandatory:"false" json:"sourceDetails"`
+
+	// Free form tags applied to this resource
+	FreeFormTags map[string]string `mandatory:"false" json:"freeformTags,omitempty"`
 }
 
 func (m Volume) String() string {
@@ -72,6 +75,7 @@ func (m *Volume) UnmarshalJSON(data []byte) (e error) {
 		LifecycleState     VolumeLifecycleStateEnum `json:"lifecycleState"`
 		SizeInMBs          *int                     `json:"sizeInMBs"`
 		TimeCreated        *common.SDKTime          `json:"timeCreated"`
+		FreeFormTags       map[string]string        `json:"freeformTags"`
 	}{}
 
 	e = json.Unmarshal(data, &model)
@@ -92,6 +96,7 @@ func (m *Volume) UnmarshalJSON(data []byte) (e error) {
 	m.LifecycleState = model.LifecycleState
 	m.SizeInMBs = model.SizeInMBs
 	m.TimeCreated = model.TimeCreated
+	m.FreeFormTags = model.FreeFormTags
 	return
 }
 
diff --git a/github.com/oracle/oci-go-sdk/oci.go b/github.com/oracle/oci-go-sdk/oci.go
index 47fcef9..4213d37 100644
--- a/github.com/oracle/oci-go-sdk/oci.go
+++ b/github.com/oracle/oci-go-sdk/oci.go
@@ -1,4 +1,3 @@
-
 /*
 This is the official Go SDK for Oracle Cloud Infrastructure
 
@@ -226,7 +225,7 @@ Please refer to this link: https://github.com/oracle/oci-go-sdk#help
 
 
 
- */
+*/
 package oci
 
 //go:generate go run cmd/genver/main.go cmd/genver/version_template.go --output common/version.go
